### PR TITLE
Unify Intel LLVM optimizations under lib/Target/LLVMIR

### PR DESCRIPTION
**Pull request by alexbaden**
Depends on #2775

I agree it would be nice to match upstream directory structure for our LLVM IR post processing passes - I've unified them here on top of the branch from 2775.

### CODE REVIEW

File: bin/CMakeLists.txt [hunk 1]
```diff
@@ -13,6 +13,9 @@ target_link_libraries(triton-opt PRIVATE
   TritonTransforms
   TritonGPUTransforms
   TritonNvidiaGPUTransforms
+  TritonIntelLLVMIR
+  TritonIntelGPUIR
+  TritonIntelGPUTransforms
```
Comment (whitneywhtsang) [inline bin/CMakeLists.txt:18]: They were removed intentionally in https://github.com/intel/intel-xpu-backend-for-triton/pull/2874.
```suggestion
```

Comment (alexbaden) [reply bin/CMakeLists.txt:18]: Yes I know, I was wondering if that was related to the flakey builds but it turns out the issue is here: https://github.com/intel/intel-xpu-backend-for-triton/pull/2901

Comment (alexbaden) [reply bin/CMakeLists.txt:18]: Fixed here: https://github.com/intel/intel-xpu-backend-for-triton/pull/2775

**Comment by alexbaden** [comment]
Rebased on #2775 - with this change all the Intel LLVM passes are in `third_party/intel/lib/Target/LLVMIR` (https://github.com/intel/intel-xpu-backend-for-triton/tree/alex/unify_intel_llvmir_processing/third_party/intel/lib/Target/LLVMIR)

File: bin/triton-llvm-opt.cpp [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /// Trimmed down clone of llvm opt to be able to test triton custom llvm ir
 /// passes.
 #include "lib/Target/LLVMIR/LLVMPasses.h"
-#include "third_party/intel/lib/LLVMIR/LLVMPasses.h"
+#include "third_party/intel/lib/Target/LLVMIR/LLVMPasses.h"
```
Comment (etiotto) [inline bin/triton-llvm-opt.cpp:4]: I wonder if we really need to add this to this (common) makefile, it wasn't required for the PostProcessing library.

File: third_party/intel/CMakeLists.txt [hunk 3]
```diff
@@ -9,7 +9,6 @@ add_triton_plugin(TritonXPU
 
   LINK_LIBS
   MLIRGPUToLLVMSPV
-  PostProcessLLVMIR
```
Comment (etiotto) [inline third_party/intel/CMakeLists.txt:12]: I don't understand the reason we need to remove this from the intel 3rd party makefile.

File: third_party/intel/lib/Target/LLVMIR/CMakeLists.txt [hunk 4]
```diff
@@ -1,8 +1,12 @@
 add_subdirectory(Dialect)
 
-add_mlir_translation_library(PostProcessLLVMIR
+add_triton_library(TritonIntelLLVMIR
```
Comment (etiotto) [inline third_party/intel/lib/Target/LLVMIR/CMakeLists.txt:3]: What is the reason for not using `add_mlir_translation_library` and instead using `add_triton_library`

Comment (alexbaden) [reply third_party/intel/lib/Target/LLVMIR/CMakeLists.txt:3]: Consistency with upstream - all libraries use `add_triton_library` and not `add_mlir_translation_library`. Semantically I don't think there are too many differences, though I do not see where the DEPENDS section is processed if you have a `mlir_translation_library`.

File: bin/triton-llvm-opt.cpp (hunk 2 above)
Comment (alexbaden) [reply bin/triton-llvm-opt.cpp:4]: PostProcessing does not test passes via lit test, but the FreezeMaskedLayout pass (which is now part of the same library) does.

File: third_party/intel/lib/Target/LLVMIR/CMakeLists.txt (hunk 4 above)
Comment (alexbaden) [reply third_party/intel/lib/Target/LLVMIR/CMakeLists.txt:3]: Actually I tested this again and we need `add_triton_library` to properly add the library to the Triton runtime for linking. With `add_mlir_translation_library` I get link errors for the LLVM IR pass.

File: third_party/intel/CMakeLists.txt (hunk 3 above)
Comment (alexbaden) [reply third_party/intel/CMakeLists.txt:12]: `add_triton_library` adds the lib to `TRITON_LIBS` so it is already linked, no need to add it again here.

**Review by whitneywhtsang — APPROVED** [review]

**Pull request merged**